pkg/plugin/validation.go

```diff
@@ -17,32 +17,62 @@ limitations under the License.
 package plugin
 
 import (
-	"errors"
 	"fmt"
 
 	"github.com/blang/semver"
 
 	"sigs.k8s.io/kubebuilder/pkg/internal/validation"
 )
 
+type errInvalidPluginVersion struct {
```

Comment:
Add new plugin validations shows not be part of the scope `plugins: abstract base Go plugin for easier version upgrades `

Reply:
See #1541.